packages/plugin-npm-cli/sources/commands/npm/publish.ts

```diff
@@ -50,13 +50,16 @@ export default (clipanion: any, pluginConfiguration: PluginConfiguration) => cli
     const ident = workspace.manifest.name;
     const version = workspace.manifest.version;
 
+    const publishRegistry = workspace.manifest.publishConfig && workspace.manifest.publishConfig.registry || configuration.get('npmPublishRegistry') as string|undefined;
+
     const report = await StreamReport.start({configuration, stdout}, async report => {
       // Not an error if --tolerate-republish is set
       if (tolerateRepublish) {
         try {
           const registryData = await npmHttpUtils.get(npmHttpUtils.getIdentUrl(ident), {
             configuration,
             ident,
+            registry: publishRegistry,
```

Comment:
I'm not fond of passing the `registry` as-is (it puts the burden on the caller to select the right one, which will likely be the same logic over and over). What do you think of instead passing `workspace` (or maybe `workspace.manifest`) and letting `npmHttpUtils` derive the right registry?

Reply:
That would make npmHttpUtils aware of whether we're publishing or just accessing the regular registry. I'm not sure how to write that into code without adding two options: a boolean or enum to say we're publishing and the workspace instance. Two options looks bad to me, but maybe that's just me

Reply:
Aaaah right 🤔 I was thinking that we could add this logic to `put` only, but then I realized that we also use `get` with `--tolerate-republish`. Okay then!